### PR TITLE
Refactor distribution buckets (with extra state).

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
#### Changes
- updated distribution bucket storage (from distribution type family container to a double map)
- introduced a new type - `DistributionBucketState`
- types

#### Review comments
- This is the second option for the distribution bucket family changes ([this is the first one](https://github.com/Joystream/joystream/pull/2792))

### CODE REVIEW

File: types/src/storage.ts [hunk 1]
```diff
@@ -220,13 +220,23 @@ export class DistributionBucket
   })
   implements IDistributionBucket {}
 
+export type IDistributionBucketState = {
+  accepting_new_bags: bool
+}
+
+export class DistributionBucketState
```
Comment (Lezek123) [inline types/src/storage.ts:227]: Please add the new type to `storageTypes` const below and regenerate `defs.json`

**Review by Lezek123 — CHANGES_REQUESTED** [review]

File: types/src/storage.ts (hunk 1 above)
Comment (Lezek123) [reply types/src/storage.ts:227]: Also please update `chain-metadata.json` with `yarn update-chain-metadata`

Comment (shamil-gadelshin) [reply types/src/storage.ts:227]: Fixed.

**Review by Lezek123 — APPROVED** [review]
LGTM
Didn't encounter any issues during my local tests

**Review by bedeho — APPROVED** [review]
Approved, with 2nd review of core concept later.

**Comment by shamil-gadelshin** [comment]
Superseded by https://github.com/Joystream/joystream/pull/2897

**Pull request closed without merging**